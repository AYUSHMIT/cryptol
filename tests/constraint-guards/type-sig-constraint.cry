bar : {n, a} (Zero a) => [n]a
bar | n == 1 => zero
    | n != 1 => zero
