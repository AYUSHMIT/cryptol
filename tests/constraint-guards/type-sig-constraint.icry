// This is a "regression test" for
// https://github.com/GaloisInc/saw-script/issues/2043 in the sense that it
// verifies each guard in the numeric constraint guard definition contains two
// proof applications, which is a smoke test for testing that the bug is fixed.
// Full validation of the fix requires SAW, however.
:set coreLint=on
:l type-sig-constraint.cry
