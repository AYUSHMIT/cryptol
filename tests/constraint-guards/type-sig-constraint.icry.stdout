Loading module Cryptol
Loading module Cryptol
Loading module Main
module Main

/* Not recursive */
Main::bar(n == 1) : {n, a} (Zero a, n == 1) => [n]a
Main::bar(n == 1) =
  \{n, a} (Zero a, n == 1) -> Cryptol::zero ([n]a) <>

/* Not recursive */
Main::bar(n != 1) : {n, a} (Zero a, n != 1) => [n]a
Main::bar(n != 1) =
  \{n, a} (Zero a, n != 1) -> Cryptol::zero ([n]a) <>

/* Not recursive */
Main::bar : {n, a} (Zero a) => [n]a
Main::bar =
  \{n, a} (Zero a) ->
    (propguards  | n == 1 => Main::bar(n == 1) n a <> <>
     | n != 1 => Main::bar(n != 1) n a <> <>)


